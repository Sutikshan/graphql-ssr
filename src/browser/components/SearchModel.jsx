import React from "react";
import PropTypes from "prop-types";
import { graphql } from "react-apollo";
import { getModelsByMake } from "./queries";

export function SearchModel({ data, onModelSelect, modelId }) {
  let { models, loading } = data;
  if (loading) {
    models = [];
  }

  return (
    <div>
      <select onChange={onModelSelect} value={modelId}>
        <option>Select Car Model</option>
        {models.map(model => {
          return (
            <option key={model.id} value={model.id}>
              {model.name}
            </option>
          );
        })}
      </select>
    </div>
  );
}

SearchModel.propTypes = {
  data: PropTypes.shape({
    models: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string
      })
    )
  }),
  onModelSelect: PropTypes.func.isRequired,
  makeId: PropTypes.number,
  modelId: PropTypes.number
};

export default graphql(getModelsByMake)(SearchModel);
