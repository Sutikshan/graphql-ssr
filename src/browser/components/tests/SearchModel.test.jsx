import React from "react";
import { shallow } from "enzyme";
import { SearchModel } from "../SearchModel";

it("should render a SearchModel component correctly", () => {
  const data = {
    models: [{ id: 1, Name: "i30" }]
  };
  const wrapper = shallow(<SearchModel data={data} onModelSelect={() => {}} />);

  expect(wrapper).toMatchSnapshot();
});
