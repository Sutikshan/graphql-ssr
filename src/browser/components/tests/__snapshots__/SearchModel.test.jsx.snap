// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a SearchModel component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchModel
    data={
      Object {
        "models": Array [
          Object {
            "Name": "i30",
            "id": 1,
          },
        ],
      }
    }
    onModelSelect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        onChange={[Function]}
        value={undefined}
      >
        <option>
          Select Car Model
        </option>
        <option
          value={1}
        />
      </select>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option>
            Select Car Model
          </option>,
          Array [
            <option
              value={1}
            />,
          ],
        ],
        "onChange": [Function],
        "value": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select Car Model",
          },
          "ref": null,
          "rendered": "Select Car Model",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "value": 1,
          },
          "ref": null,
          "rendered": null,
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          onChange={[Function]}
          value={undefined}
        >
          <option>
            Select Car Model
          </option>
          <option
            value={1}
          />
        </select>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option>
              Select Car Model
            </option>,
            Array [
              <option
                value={1}
              />,
            ],
          ],
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Select Car Model",
            },
            "ref": null,
            "rendered": "Select Car Model",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "value": 1,
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
